flume-ng-core/src/main/java/org/apache/flume/serialization/AbstractAvroEventDeserializer.java

```diff
@@ -0,0 +1,179 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.flume.serialization;
+
+import com.google.common.annotations.VisibleForTesting;
+import com.google.common.collect.Lists;
+import org.apache.avro.Schema;
+import org.apache.avro.SchemaNormalization;
+import org.apache.avro.file.DataFileConstants;
+import org.apache.avro.file.DataFileReader;
+import org.apache.avro.file.SeekableInput;
+import org.apache.avro.generic.GenericDatumReader;
+import org.apache.avro.generic.GenericDatumWriter;
+import org.apache.avro.generic.GenericRecord;
+import org.apache.avro.io.BinaryEncoder;
+import org.apache.avro.io.EncoderFactory;
+import org.apache.commons.codec.binary.Hex;
+import org.apache.flume.Event;
+import org.apache.flume.event.EventBuilder;
+
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.security.NoSuchAlgorithmException;
+import java.util.List;
+
+/**
+ * A deserializer that parses Avro container files, generating one Flume event
+ * per record in the Avro file, and storing binary avro-encoded records in
+ * the Flume event body.
+ */
+public abstract class AbstractAvroEventDeserializer implements EventDeserializer {
+
+  protected AvroSchemaType schemaType;
+  protected ResettableInputStream ris;
+
+  protected Schema schema;
+  private byte[] schemaHash;
+  private String schemaHashString;
+  private DataFileReader<GenericRecord> fileReader;
+  private GenericDatumWriter datumWriter;
+  protected GenericRecord record;
+  private ByteArrayOutputStream out;
+  private BinaryEncoder encoder;
+
+  @VisibleForTesting
+  public static enum AvroSchemaType {
+    HASH,
+    LITERAL;
+  }
+
+  public static final String CONFIG_SCHEMA_TYPE_KEY = "schemaType";
+  public static final String AVRO_SCHEMA_HEADER_HASH
+      = "flume.avro.schema.hash";
+  public static final String AVRO_SCHEMA_HEADER_LITERAL
+      = "flume.avro.schema.literal";
+
+
+  protected void initialize(ResettableInputStream ris)
```

Comment:
The parameter is not needed if `AbstractAvroEventDeserializer.ris` is changed to `private final`.